ecosystem/helper/src/client/utils/path.ts

```diff
@@ -0,0 +1,47 @@
+import type { RouteLocationNormalizedLoaded } from 'vue-router'
+
+const HASH_REGEXP = /#.*$/u
+
+/**
+ * Get hash from path
+ *
+ * @param path link path
+ * @returns hash
+ */
+export const getHash = (path: string): string => {
+  const match = HASH_REGEXP.exec(path)
+
+  if (match) return match[0]
+
+  return ''
+}
+
+export const normalizePath = (path: string): string =>
+  decodeURI(path)
+    .replace(HASH_REGEXP, '')
+    .replace(/(index)?\.(md|html)$/, '')
+
+/**
+ * Whether the lick is active
+ *
+ * @param route Current route
+ * @param link link path
+ * @returns Whether the lick is active
+ */
+export const isActiveLink = (
```

Comment:
This will be useful in many places, such as catalog, toc component or any comopnent which has active state with links. so this is extracted from theme-default